hoodie-common/src/main/java/com/uber/hoodie/common/table/log/block/HoodieCorruptBlock.java

```diff
@@ -26,21 +34,52 @@
 
   private final byte[] corruptedBytes;
 
-  private HoodieCorruptBlock(byte[] corruptedBytes) {
+  private final Map<HoodieMetadataType, String> metadata;
+
+  private HoodieCorruptBlock(byte[] corruptedBytes, Map<HoodieMetadataType, String> metadata) {
     this.corruptedBytes = corruptedBytes;
+    this.metadata = metadata;
+  }
+
+  private HoodieCorruptBlock(byte[] corruptedBytes) {
+    this(corruptedBytes, null);
   }
 
   @Override
   public byte[] getBytes() throws IOException {
-    return corruptedBytes;
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
likewise, we should be able to reuse these two lines across these classes, if we had LogBlock as an abstract class